test/test_tensordict.py

```diff
@@ -1443,7 +1443,7 @@ def test_tensordict_set(self, td_name, device):
         val2 = np.zeros(shape=(4, 3, 2, 1, 10))
         td.set_("key1", val2)
         assert (td.get("key1") == 0).all()
-        with pytest.raises((KeyError, AttributeError)):
+        with pytest.raises((KeyError, AttributeError, ValueError)):
```

Comment:
This test feels pretty loose at this point. It's possible e.g. something else goes wrong and a different `ValueError` gets raised. Should we limit the possibilities and maybe add a `match` argument?

Reply:
Solved